include/head/core/presence_sensing_driver.hpp

```diff
@@ -26,8 +96,42 @@ class PresenceSensingDriver {
         return voltage_read;
     }
 
+    auto static get_tool(adc::MillivoltsReadings reading) -> AttachedTool {
+        auto at = AttachedTool{.z_motor = can_ids::ToolType::UNDEFINED_TOOL,
+                               .a_motor = can_ids::ToolType::UNDEFINED_TOOL,
+                               .gripper = can_ids::ToolType::UNDEFINED_TOOL};
+        for (auto& element : OT3ToolList) {
+            if ((reading.z_motor < element.detection_upper_bound) &&
```

Comment:
If the `Tool` class had a method like `within_bounds` or something then we wouldn't have to duplicate this logic 3 times.